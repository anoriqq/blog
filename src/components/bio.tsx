import React from 'react';
import { StaticQuery, graphql, Link } from 'gatsby';
import Image from 'gatsby-image';

import { rhythm } from '../utils/typography';

export const PureBio = ({ data }) => (
  <div
    style={{
      display: `flex`,
      marginBottom: rhythm(2.5),
    }}
  >
    <Image
      fixed={data.avatar.childImageSharp.fixed}
      alt={data.site.siteMetadata.author.name}
      style={{
        marginRight: rhythm(1 / 2),
        marginBottom: 0,
        minWidth: 50,
        borderRadius: `100%`,
      }}
      imgStyle={{
        borderRadius: `50%`,
      }}
    />
    <p>
      Written by
      <strong>{` ${data.site.siteMetadata.author.name} `}</strong>
      {data.site.siteMetadata.author.summary}
      {` `}
      <Link to="/about">About</Link>
      {` `}
      <a href={`https://github.com/${data.site.siteMetadata.social.github}`}>
        GitHub
      </a>
      {` `}
      <a href={`https://twitter.com/${data.site.siteMetadata.social.twitter}`}>
        Twitter
      </a>
    </p>
  </div>
);

export const Bio = () => (
  <StaticQuery
    query={graphql`
      query BioQuery {
        avatar: file(absolutePath: { regex: "/profile-pic.jpg/" }) {
          childImageSharp {
            fixed(width: 50, height: 50) {
              ...GatsbyImageSharpFixed
            }
          }
        }
        site {
          siteMetadata {
            author {
              name
              summary
            }
            social {
              twitter
              github
            }
          }
        }
      }
    `}
    render={data => <PureBio data={data} />}
  />
);
