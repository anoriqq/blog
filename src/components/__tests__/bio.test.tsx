import React from 'react';
import renderer from 'react-test-renderer';

import { PureBio as Bio } from '../bio';

describe('Bio', () => {
  it('renders correctly', () => {
    const data = {
      avatar: {
        childImageSharp: {
          fixed: {
            base64:
              'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQADAAAAAAAAAAAAAAAAAAECAwX/xAAWAQEBAQAAAAAAAAAAAAAAAAAAAQL/2gAMAwEAAhADEAAAAe/GFztRNAUH/8QAGBAAAgMAAAAAAAAAAAAAAAAAAAEQESD/2gAIAQEAAQUCLhiz/8QAFREBAQAAAAAAAAAAAAAAAAAAESD/2gAIAQMBAT8BY//EABQRAQAAAAAAAAAAAAAAAAAAACD/2gAIAQIBAT8BH//EABQQAQAAAAAAAAAAAAAAAAAAADD/2gAIAQEABj8CH//EABoQAQABBQAAAAAAAAAAAAAAABEAARAgITH/2gAIAQEAAT8hry5mokx//9oADAMBAAIAAwAAABBEzwD/xAAVEQEBAAAAAAAAAAAAAAAAAAABIP/aAAgBAwEBPxBQx//EABYRAQEBAAAAAAAAAAAAAAAAAAEgIf/aAAgBAgEBPxAMj//EABwQAQABBAMAAAAAAAAAAAAAAAEhABFBYRAgMf/aAAgBAQABPxBIkLtEpgib54Cyaq5gdnnX/9k=',
            width: 50,
            height: 50,
            src:
              '/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg',
            srcSet:
              '/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg 1x,\n/static/897dc7abe19cc42b831a745c4a64d44d/aba1d/profile-pic.jpg 1.5x,\n/static/897dc7abe19cc42b831a745c4a64d44d/b315d/profile-pic.jpg 2x',
          },
        },
      },
      site: {
        siteMetadata: {
          author: {
            name: 'anoriqq',
            summary:
              'Web app developer in Tokyo. Interested in Node.js and TypeScript.',
          },
          social: {
            twitter: 'anoriqq',
            github: 'anoriqq',
          },
        },
      },
    };
    const tree = renderer.create(<Bio data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
