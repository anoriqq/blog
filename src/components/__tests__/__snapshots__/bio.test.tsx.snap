// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginBottom": "4.375rem",
    }
  }
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "borderRadius": "100%",
        "display": "inline-block",
        "height": 50,
        "marginBottom": 0,
        "marginRight": "0.875rem",
        "minWidth": 50,
        "overflow": "hidden",
        "position": "relative",
        "width": 50,
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQADAAAAAAAAAAAAAAAAAAECAwX/xAAWAQEBAQAAAAAAAAAAAAAAAAAAAQL/2gAMAwEAAhADEAAAAe/GFztRNAUH/8QAGBAAAgMAAAAAAAAAAAAAAAAAAAEQESD/2gAIAQEAAQUCLhiz/8QAFREBAQAAAAAAAAAAAAAAAAAAESD/2gAIAQMBAT8BY//EABQRAQAAAAAAAAAAAAAAAAAAACD/2gAIAQIBAT8BH//EABQQAQAAAAAAAAAAAAAAAAAAADD/2gAIAQEABj8CH//EABoQAQABBQAAAAAAAAAAAAAAABEAARAgITH/2gAIAQEAAT8hry5mokx//9oADAMBAAIAAwAAABBEzwD/xAAVEQEBAAAAAAAAAAAAAAAAAAABIP/aAAgBAwEBPxBQx//EABYRAQEBAAAAAAAAAAAAAAAAAAEgIf/aAAgBAgEBPxAMj//EABwQAQABBAMAAAAAAAAAAAAAAAEhABFBYRAgMf/aAAgBAQABPxBIkLtEpgib54Cyaq5gdnnX/9k="
      style={
        Object {
          "borderRadius": "50%",
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transitionDelay": "500ms",
          "width": "100%",
        }
      }
    />
    <picture>
      <source
        srcSet="/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg 1x,
/static/897dc7abe19cc42b831a745c4a64d44d/aba1d/profile-pic.jpg 1.5x,
/static/897dc7abe19cc42b831a745c4a64d44d/b315d/profile-pic.jpg 2x"
      />
      <img
        alt="anoriqq"
        height={50}
        loading="lazy"
        onLoad={[Function]}
        src="/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg"
        srcSet="/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg 1x,
/static/897dc7abe19cc42b831a745c4a64d44d/aba1d/profile-pic.jpg 1.5x,
/static/897dc7abe19cc42b831a745c4a64d44d/b315d/profile-pic.jpg 2x"
        style={
          Object {
            "borderRadius": "50%",
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
        width={50}
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg 1x,
/static/897dc7abe19cc42b831a745c4a64d44d/aba1d/profile-pic.jpg 1.5x,
/static/897dc7abe19cc42b831a745c4a64d44d/b315d/profile-pic.jpg 2x\\" /><img loading=\\"lazy\\" width=\\"50\\" height=\\"50\\" srcset=\\"/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg 1x,
/static/897dc7abe19cc42b831a745c4a64d44d/aba1d/profile-pic.jpg 1.5x,
/static/897dc7abe19cc42b831a745c4a64d44d/b315d/profile-pic.jpg 2x\\" src=\\"/static/897dc7abe19cc42b831a745c4a64d44d/99438/profile-pic.jpg\\" alt=\\"anoriqq\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
  <p>
    Written by
    <strong>
       anoriqq 
    </strong>
    Web app developer in Tokyo. Interested in Node.js and TypeScript.
     
    <a
      href="/about"
    >
      About
    </a>
     
    <a
      href="https://github.com/anoriqq"
    >
      GitHub
    </a>
     
    <a
      href="https://twitter.com/anoriqq"
    >
      Twitter
    </a>
  </p>
</div>
`;
